workflow-manager/kubernetes/kubernetes.go

```diff
@@ -25,6 +30,29 @@ type Client struct {
 // it is not empty. If dryRun is true, then any destructive API calls will be
 // made with DryRun: All.
 func NewClient(namespace string, kubeconfigPath string, dryRun bool) (*Client, error) {
+	// Making sure endpoint is available before returning kubernetes client
```

Comment:
Instead of polling for availability here, I'd prefer if we retried the call to `c.client.BatchV1.Jobs().List()` in `LostJobs()`, though only if the call fails due to the context deadline being exceeded. I think that would make us resilient to more kinds of failures, and also would rely less on implementation details of the Kubernetes client library.

Reply:
I see a challenge if we do it LostJobs() 
(1) this is purely to discover if the network is up and this needs to be done only once (when we get the client)
(2) The behaviour I discovered is that the call times out even if the network becomes available within the 30sec of the context deadline. Meaning we should use different timeouts which makes this complex and hard to maintain or very slow as an alternative. I'd prefer leaving this here. We can add more retry logic independently if to LostJobs if needed.

Reply:
Robert convinced me in Slack that since we'll soon delete this module anyway (because `workflow-manager` won't need to list Kubernetes jobs at all once the migration from the old mode is complete), we can live with this as it is.